server/integration/event-logger/src/main/java/org/infinispan/server/eventlogger/ServerEventLogger.java

```diff
@@ -89,26 +99,35 @@ TimeService getTimeService() {
    }
 
    @Override
-   public List<EventLog> getEvents(int start, int count, Optional<EventLogCategory> category, Optional<EventLogLevel> level) {
-      QueryBuilder query = getQueryFactory().from(ServerEventImpl.class).orderBy("when", SortOrder.DESC).maxResults(count).startOffset(start);
-      if (category.isPresent()) {
-         if (level.isPresent()) {
-            query.having("category").eq(category.get()).and().having("level").eq(level.get());
-         } else {
-            query.having("category").eq(category.get());
+   public List<EventLog> getEvents(Instant start, int count, Optional<EventLogCategory> category, Optional<EventLogLevel> level) {
+      List<EventLog> events = new ArrayList<>();
+      ConcurrentMap<Address, Throwable> throwables = CollectionFactory.makeConcurrentMap();
```

Comment:
Maybe just use a ConcurrentLinkedQueue instead?  Since we don't ever use the keys.